src/prop13/graphlibrary/Edge.java

```diff
@@ -5,13 +5,33 @@
  */
 public abstract class Edge
 {
-    protected Node n1;
-    protected Node n2;
-    protected int weight;
+    private Node n1;
+    private Node n2;
+    private int weight;
 
-    public Edge()
-    {
-    }
+	public Node getN1() {
+		return n1;
+	}
+
+	public Node getN2() {
+		return n2;
+	}
+
+	public int getWeight() {
+		return weight;
+	}
+
+	public void setN1(Node n1) {
```

Comment:
I am not sure it's a good idea to let people change one of the nodes of an edge, it is not a needed functionality of the class and it is really problematic for the graph class (you could change one of the nodes of an edge without making any change to the graph). I think that the nodes of an edge shoud be setted with the constructor and impossible to change.